component/otelcol/processor/resourcedetection/resourcedetection.go

```diff
@@ -0,0 +1,223 @@
+
+	// Azure contains user-specified configurations for the azure detector
+	AzureConfig azure.Config `river:"azure,block,optional"`
+
+	// Aks contains user-specified configurations for the aks detector
+	AksConfig aks.Config `river:"aks,block,optional"`
+
+	// ConsulConfig contains user-specified configurations for the Consul detector
+	ConsulConfig consul.Config `river:"consul,block,optional"`
+
+	// DockerConfig contains user-specified configurations for the docker detector
+	DockerConfig docker.Config `river:"docker,block,optional"`
+
+	// GcpConfig contains user-specified configurations for the gcp detector
+	GcpConfig gcp.Config `river:"gcp,block,optional"`
+
+	// HerokuConfig contains user-specified configurations for the heroku detector
+	HerokuConfig heroku.Config `river:"heroku,block,optional"`
+
+	// SystemConfig contains user-specified configurations for the System detector
+	SystemConfig system.Config `river:"system,block,optional"`
+
+	// OpenShift contains user-specified configurations for the Openshift detector
+	OpenShiftConfig openshift.Config `river:"openshift,block,optional"`
+
+	// KubernetesNode contains user-specified configurations for the K8SNode detector
+	KubernetesNodeConfig kubernetes_node.Config `river:"kubernetes_node,block,optional"`
+}
+
+var (
+	_ processor.Arguments = Arguments{}
+	_ river.Validator     = (*Arguments)(nil)
+	_ river.Defaulter     = (*Arguments)(nil)
+)
+
+// DefaultArguments holds default settings for Arguments.
+var DefaultArguments = Arguments{
+	Detectors: []string{"env"},
+	Override:  true,
+	Timeout:   5 * time.Second,
+	DetectorConfig: DetectorConfig{
+		EC2Config:              ec2.DefaultArguments,
+		ECSConfig:              ecs.DefaultArguments,
+		EKSConfig:              eks.DefaultArguments,
+		ElasticbeanstalkConfig: elasticbeanstalk.DefaultArguments,
+		LambdaConfig:           lambda.DefaultArguments,
+		AzureConfig:            azure.DefaultArguments,
+		AksConfig:              aks.DefaultArguments,
+		ConsulConfig:           consul.DefaultArguments,
+		DockerConfig:           docker.DefaultArguments,
+		GcpConfig:              gcp.DefaultArguments,
+		HerokuConfig:           heroku.DefaultArguments,
+		SystemConfig:           system.DefaultArguments,
+		OpenShiftConfig:        openshift.DefaultArguments,
+		KubernetesNodeConfig:   kubernetes_node.DefaultArguments,
+	},
+}
+
+// SetToDefault implements river.Defaulter.
+func (args *Arguments) SetToDefault() {
+	*args = DefaultArguments
+}
+
+// Validate implements river.Validator.
+func (args *Arguments) Validate() error {
+	if len(args.Detectors) == 0 {
+		return fmt.Errorf("at least one detector must be specified")
+	}
+	return nil
+}
+
+func (args Arguments) ConvertDetectors() []string {
+	if args.Detectors == nil {
+		return nil
+	}
+
+	res := make([]string, 0, len(args.Detectors))
+	for _, detector := range args.Detectors {
+		//TODO(ptodev): Check if the detector name is valid
```

Comment:
Thanks for spotting this! I think the Collector code would throw an error, but it would be more correct for River to do it in the Validate() function. I updated the code to do just that, and added tests.